showcase/go/showcase_integration_test.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package showcase_integration
+
+import (
+	"context"
+	"flag"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	showcase "cloud.google.com/go/showcase/apiv1alpha2"
+	durationpb "github.com/golang/protobuf/ptypes/duration"
+	genprotopb "github.com/googleapis/gapic-showcase/server/genproto"
+	"google.golang.org/api/option"
+	spb "google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var client *showcase.EchoClient
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	conn, err := grpc.Dial("localhost:7469", grpc.WithInsecure())
+	if err != nil {
+		log.Fatal(err)
+	}
+	clientOpt := option.WithGRPCConn(conn)
+	client, err = showcase.NewEchoClient(context.Background(), clientOpt)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestEcho(t *testing.T) {
+	content := "hello world!"
+	req := &genprotopb.EchoRequest{
+		Response: &genprotopb.EchoRequest_Content{
+			Content: content,
+		},
+	}
+	resp, err := client.Echo(context.Background(), req)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp.GetContent() != req.GetContent() {
+		t.Errorf(
+			"Echo did not receive expected value. Got %s, Wanted %s",
```

Comment:
Not a serious issue, though the messages can be shorter like `Echo() = %q, want %q`.

I'd rather use `%q`. It's unlikely but error messages with `%s` will be really confusing if we have unprintable characters.

Reply:
Done!